compiler/rustc_middle/src/hir/map/mod.rs

```diff
@@ -155,19 +155,22 @@ impl<'hir> Map<'hir> {
 
     #[inline]
     pub fn definitions(&self) -> &'hir Definitions {
-        &self.tcx.definitions
+        // Accessing the definitions is ok, since all its contents are tracked by the query system.
```

Comment:
Can you elaborate this? I'm not sure where that tracking happens.

Reply:
Now you ask, I have a doubt this is true for *all* fields in `definitions`. The `LocalDefId <-> DefPath/DefPathHash/DefKey` maps are stable by construction. The stability of the `LocalDefId <-> HirId` maps is related to #85266.
The other fields may not be stable :frowning_face: